### DISCUSSION TITLE
Not connected to <ip address>. Timed out while waiting for handshake.

### DISCUSSION THREAD

**Discussion by ravisankar-PIO** [codefori/vscode-ibmi#1537] [Q&A] [answered]
Hi,
   I am getting a ```Timed out while waiting for handshake``` error while connecting to a V7R2 system. 
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/139198015/12af4217-92eb-48d7-8b00-9d54ddefe376)

**Note** I am connected to VPN.

```NETSTAT *CNN``` tells me that SSH daemon is started. 
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/139198015/8abb577b-3a7f-414b-b301-4c0ed71c1597)

Strangely, I am able to spawn a SSH Terminal session using ACS. 
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/139198015/233d9b86-a997-495a-908f-d6f3066d0e37)

But I am not able to create an SSH connection using putty or command prompt. I think it is getting timed out. 

**Do you think I need to enable/disable something in the IBM i in order to allow SSH connection from other clients?**

Thank you,
Ravi.

<hr />

⚠️ **REMOVE THIS LINE AND ANY SENSITIVE INFORMATION BELOW!** ⚠️

|Context|Version|
|-|-|
|Code for IBM i version|2.2.0|
|Visual Studio Code version|1.82.0|
|Operating System|win32_x64|

<details>
<summary>Active extensions</summary>

```
COBOL (cobol): 9.8.31
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.3.1
Db2 for IBM i (vscode-db2i): 0.4.1
Emmet (emmet): 1.0.0
Error Lens (errorlens): 3.13.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
GitHub (github): 0.0.1
GitHub Authentication (github-authentication): 0.0.2
IBM i Notebooks (vscode-ibmi-notebooks): 0.0.6
JSON Language Features (json-language-features): 1.0.0
Merge Conflict (merge-conflict): 1.0.0
Microsoft Account (microsoft-authentication): 0.0.1
Node Debug Auto-attach (debug-auto-launch): 1.0.0
RPGLE (vscode-rpgle): 0.21.4
TODO Highlight (vscode-todo-highlight): 1.0.5
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
WSL: Recommender (remote-wsl-recommender): 0.0.19
```
</details>

<hr />

**_Not connected_** 🔌

**Comment by ravisankar-PIO** [comment]
Hi Volunteers! Please let me know if you have any inputs on this? 

Regards,
Ravi

**Comment by chrjorgensen** [comment]
@ravisankar-PIO This is a network issue, and I would suggest you talk to your network administrators about what traffic is allowed - seems like a firewall is blocking your connection attempts from VPN.

There is no special requirements for running SSH on IBM i. However, it is advisable to follow best practice and make the SSH server more secure - see this from IBM i Security Specialist Carol Woodbury: https://www.mcpressonline.com/security/ibm-i-os400-i5os/controlling-access-to-ssh-on-ibm-i

**Comment by ravisankar-PIO** [comment, marked as answer]
Figured it out!

The problem is that I was trying to SSH using the port# 23. Whereas the the port# for SSH is 22. It worked for me after I changed the port number to 22 in the login settings of the IBM i connection. 

Thank you for the prompt response!
